tool/tsh/tsh.go

```diff
@@ -1930,28 +1931,70 @@ func (l nodeListings) Swap(i, j int) {
 }
 
 func listNodesAllClusters(cf *CLIConf) error {
-	var listings nodeListings
+	// Fetch database listings for profiles in parallel. Set an arbitrary limit
+	// just in case.
+	group, groupCtx := errgroup.WithContext(cf.Context)
+	group.SetLimit(4)
+
+	nodeListingsResultChan := make(chan nodeListings)
+	nodeListingsCollectChan := make(chan nodeListings)
+	go func() {
+		var nodeListings nodeListings
+		for {
+			select {
+			case items := <-nodeListingsResultChan:
+				nodeListings = append(nodeListings, items...)
+			case <-groupCtx.Done():
+				nodeListingsCollectChan <- nodeListings
```

Comment:
nit: declaring a variable of the same name as the type is a little confusing, and in go it shadows the type.
```suggestion
		var nodeListings nodeListings
		for {
			select {
			case items := <-nodeListingsResultChan:
				listings = append(listings, items...)
			case <-groupCtx.Done():
				nodeListingsCollectChan <- listings
```